lib/datadog/kit/identity.rb

```diff
@@ -56,9 +57,17 @@ def self.set_user(trace, id:, email: nil, name: nil, session_id: nil, role: nil,
         others.each do |k, v|
           trace.set_tag("usr.#{k}", v) unless v.nil?
         end
+
+        if ::Datadog::AppSec.settings.enabled
+          _, appsec_response = ::Datadog::AppSec::Instrumentation.gateway.push('identity.set_user', id)
```

Comment:
I see your point of passing an object or a struct, but at the same time, it feels like the early decision to pass an object when we only care about a single attribute of the thing might not be necessary. Delaying that internal decision an internal thing is a good idea. 

Also, the Gateway does not validate what the data is when pushed, so in a way, it could be anything, an object, a string...

Reply:
Having it be an object makes it easier to track the expectations and usage from the sender to the receiver. It also encapsulates the extensibility of it

In addition it forces us to "find a place" to define the matching class, which was not the case previously since they were predefined `Rack::Request` and `Rack::Response`

... which I was wondering about whether they should be boxed into some gateway-specific object. Something like (very rough, pseudocode, names not exact, jus to give an idea of the hierarchy):

```
Gateway::Object
Gateway::User < Gateway::Object
Gateway::Request < Gateway::Object
Gateway::Response < Gateway::Object
```

and then we can type (again, rough pseudocode to get the idea):
- `Gateway#push: (address, Gateway::Object) -> ...`
- `Gateway#watch: () { |Gateway::Object| }`